cypress/support/commands.js

```diff
@@ -24,6 +26,44 @@
 // -- This will overwrite an existing command --
 // Cypress.Commands.overwrite("visit", (originalFn, url, options) => { ... })
 
+/**
+ * Use this function to sign a user out without lingering network calls causing unintended side-effects.
+ */
+Cypress.Commands.add('signOutUser', () => {
+  const intercepts = getInterceptsForLingeringUserRequests(false);
+
+  return cy.request('DELETE', '/users/sign_out').then(() => {
+    cy.visit('/');
+    cy.wait(intercepts);
+  });
+});
+
+/**
+ * Logins in a user and visits the given URL, waiting for all user-related network requests triggered by the login to complete.
+ * This ensures that no user side effects bleed into subsequent tests.
+ */
+Cypress.Commands.add('loginAndVisit', (user, url) => {
+  cy.loginUser(user).then(() => {
+    cy.visitAndWaitForUserSideEffects(url);
+  });
+});
+
+/**
+ * Visits the given URL, waiting for all user-related network requests to complete.
+ * This ensures that no user side effects bleed into subsequent tests.
+ */
+Cypress.Commands.add('visitAndWaitForUserSideEffects', (url, options) => {
```

Comment:
This is offered as a command separate to `loginAndVisit` because in some cases we do not want to visit a URL directly after login, but instead complete some other test setup (e.g. creating an article via POST request)